app/tests/spec/models/account.js

```diff
@@ -650,6 +650,26 @@ function (chai, sinon, p, Constants, Assertion, ProfileClient,
       });
     });
 
+    describe('fetchProfile', function () {
+      it('returns profile', function () {
+        var name = 'snoopy';
+        sinon.stub(account, 'getProfile', function () {
+          return p({ avatar: PNG_URL, displayName: name });
+        });
+
+        sinon.spy(account, 'setProfileImage');
+
+        return account.fetchProfile()
+          .then(function () {
+            assert.equal(account.get('profileImageUrl'), PNG_URL);
+            assert.isTrue(account.setProfileImage.called);
+            assert.equal(account.setProfileImage.args[0][0].get('url'), PNG_URL);
+            assert.isTrue(account.get('hadProfileImageSetBefore'));
```

Comment:
It's a flag we set so we know the account has opted in to avatars. It's how
we gracefully enable them for nightly users.
On Aug 4, 2015 5:00 AM, "Shane Tomlinson" <notifications@github.com> wrote:

> In app/tests/spec/models/account.js
> <https://github.com/mozilla/fxa-content-server/pull/2748#discussion_r36179835>
> :
>
> > @@ -650,6 +650,26 @@ function (chai, sinon, p, Constants, Assertion, ProfileClient,
> >        });
> >      });
> >
> > +    describe('fetchProfile', function () {
> > +      it('returns profile', function () {
> > +        var name = 'snoopy';
> > +        sinon.stub(account, 'getProfile', function () {
> > +          return p({ avatar: PNG_URL, displayName: name });
> > +        });
> > +
> > +        sinon.spy(account, 'setProfileImage');
> > +
> > +        return account.fetchProfile()
> > +          .then(function () {
> > +            assert.equal(account.get('profileImageUrl'), PNG_URL);
> > +            assert.isTrue(account.setProfileImage.called);
> > +            assert.equal(account.setProfileImage.args[0][0].get('url'), PNG_URL);
> > +            assert.isTrue(account.get('hadProfileImageSetBefore'));
>
> What's this field do?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/fxa-content-server/pull/2748/files#r36179835>.
>